tools/testsys-config/src/lib.rs

```diff
@@ -252,6 +252,9 @@ pub struct GenericVariantConfig {
     pub instance_type: Option<String>,
     /// Specify how Bottlerocket instances should be launched (ec2, karpenter)
     pub resource_agent_type: Option<ResourceAgentType>,
+    /// Overrides the EKS service endpoint for TestSys agents gathering EKS cluster metadata
+    /// (only for pre-existing EKS clusters, does not apply to new EKS cluster creation)
+    pub eks_service_endpoint: Option<String>,
```

Comment:
This should go in the `dev` config since it will rarely be needed and could cause confusion.